File: parking_lot_system/parking_lot/parking_spot.py
import threading

class ParkingSpot:
    def __init__(self, spot_id, spot_type, level_id):
        self.spot_id = spot_id
        self.spot_type = spot_type
        self.level_id = level_id  # Add level_id to the ParkingSpot
        self.is_occupied = False
        self.vehicle = None
        self.lock = threading.Lock()

    def park_vehicle(self, vehicle):
        with self.lock:
            if not self.is_occupied and self.spot_type == vehicle.vehicle_type:
                self.vehicle = vehicle
                self.is_occupied = True
                return True
            return False

    def remove_vehicle(self):
        with self.lock:
            self.vehicle = None
            self.is_occupied = False

File: parking_lot_system_go/parkinglot/parking_spot.go
package parkinglot

import "sync"

type ParkingSpot struct {
	SpotID     int
	SpotType   VehicleType
	LevelID    int
	IsOccupied bool
	Vehicle    Vehicle
	sync.Mutex
}

func NewParkingSpot(spotID int, spotType VehicleType, levelID int) *ParkingSpot {
	return &ParkingSpot{
		SpotID:   spotID,
		SpotType: spotType,
		LevelID:  levelID,
	}
}

func (ps *ParkingSpot) ParkVehicle(vehicle Vehicle) bool {
	ps.Lock()
	defer ps.Unlock()
	if !ps.IsOccupied && ps.SpotType == vehicle.GetType() {
		ps.Vehicle = vehicle
		ps.IsOccupied = true
		return true
	}
	return false
}

func (ps *ParkingSpot) RemoveVehicle() {
	ps.Lock()
	defer ps.Unlock()
	ps.Vehicle = nil
	ps.IsOccupied = false
}
